### PR TITLE
Noseam has been refactored to be callable. Makefile test has been converted to a gtest and removed.

### PR DESCRIPTION
**Pull request by kledmundson**
## Description

The _noseam_ application has been refactored to be callable. The existing Makefile test has been converted to a gtest utilizing the ThreeImageNetwork test fixture. Additional tests have been added. The Makefile test has been removed.

## Related Issue

https://github.com/DOI-USGS/ISIS3/issues/5599

## How Has This Been Validated?

ctest suite results...

ctest -R Noseam --output-on-failure
Test project /Users/kledmundson/ISISDev/noseam/Jul182024/ISIS3/build
    Start 1255: ThreeImageNetwork.FunctionalTestNoseamDefault
1/5 Test #1255: ThreeImageNetwork.FunctionalTestNoseamDefault ................   Passed    5.65 sec
    Start 1256: ThreeImageNetwork.FunctionalTestNoseamEvenBoxFilterSamples
2/5 Test #1256: ThreeImageNetwork.FunctionalTestNoseamEvenBoxFilterSamples ...   Passed    2.82 sec
    Start 1257: ThreeImageNetwork.FunctionalTestNoseamEvenBoxFilterLines
3/5 Test #1257: ThreeImageNetwork.FunctionalTestNoseamEvenBoxFilterLines .....   Passed    2.85 sec
    Start 1258: ThreeImageNetwork.FunctionalTestNoseamNoBoxFilterSamples
4/5 Test #1258: ThreeImageNetwork.FunctionalTestNoseamNoBoxFilterSamples .....   Passed    1.65 sec
    Start 1259: ThreeImageNetwork.FunctionalTestNoseamNoBoxFilterLines
5/5 Test #1259: ThreeImageNetwork.FunctionalTestNoseamNoBoxFilterLines .......   Passed    1.66 sec

100% tests passed, 0 tests failed out of 5

Total Test time (real) =  14.83 sec

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Infrastructure change (changes to things like CI or the build system that do not impact users)

## Checklist:

- [X] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [X] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [X] I have added my user impacting change to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document. 

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [X] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Comment by amystamile-usgs** [comment]
Two test failures for this PR:
	1255 - ThreeImageNetwork.FunctionalTestNoseamDefault (Failed)
	1256 - ThreeImageNetwork.FunctionalTestNoseamEvenBoxFilterSamples (Failed)

File: isis/tests/FunctionalTestsNoseam.cpp
Comment (4353jami) [inline isis/tests/FunctionalTestsNoseam.cpp:1]: 5606

**Review by 4353jami — APPROVED** [review]

**Review by 4353jami — APPROVED** [review]

**Review by amystamile-usgs — CHANGES_REQUESTED** [review]
Tests need to be fixed before approval.

**Review by acpaquette — DISMISSED** [review]
Tests looked good, just need approval from @amystamile-usgs

**Comment by acpaquette** [comment]
@kledmundson Change log needs to be updated here as well

**Comment by amystamile-usgs** [comment]
👍 Looks good to me. Just needs merge conflicts fixed before approving.

**Review by acpaquette — DISMISSED** [review]

**Comment by acpaquette** [comment]
@kledmundson Looks like one last merge conflict needs to be resolved here

**Review by amystamile-usgs — APPROVED** [review]

**Pull request merged**
